packages/framework/src/Model/Product/Collection/ProductUrlsBatchLoader.php

```diff
@@ -60,11 +56,11 @@ public function getProductUrl(Product $product, DomainConfig $domainConfig): str
     {
         $key = $this->getKey($product, $domainConfig);
 
-        if (!array_key_exists($key, $this->loadedProductUrls)) {
+        if (!$this->localCacheFacade->hasItem(static::PRODUCT_URLS_CACHE_NAMESPACE, $key)) {
```

Comment:
same suggestion as above, what about directly calling `getItem` and wrapping it in the try-catch block?